intTests/test_type_errors/err002.log.good

```diff
@@ -1,7 +1,7 @@
  Loading file "err002.saw"
  err002.saw:5:44-5:57: Type mismatch.
     Mismatch of type constructors. Expected: String but got ([])
-    internal:1:1-1:7: The type String arises from this type annotation
+    <type of llvm_alias>:1:1-1:7: The type String arises from this type annotation
```

Comment:
I have to say, I find the way that this bit of provenance is reported to be rather awkward. It talks about the type of `llvm_alias`, but then proceeds to refer to the source position `1:1-1:7`, which is not terribly useful information for most users.

I wonder if we should just have a special case in the provenance-printing for builtins such that they're printed more like this:

```
The type String arises from the type of llvm_alias
```

This would likely require much more work than you wanted to put into this omnibus error-message-improvement PR, but I wanted to note this here since I find the new output somewhat odd. I'm happy not to fix this as part of this PR.